protocol/app/ante/market_update.go

```diff
@@ -92,10 +100,16 @@ func (d ValidateMarketUpdateDecorator) AnteHandle(
 	return next(ctx, tx, simulate)
 }
 
-func (d ValidateMarketUpdateDecorator) doMarketsContainCrossMarket(ctx sdk.Context, markets []mmtypes.Market) bool {
+// doMarketsContainRestrictedMarket checks if any of the given markets are restricted:
+// 1. markets listed as CROSS perpetuals are restricted
+// 2. the USDT/USD market is always restricted
+func (d ValidateMarketUpdateDecorator) doMarketsContainRestrictedMarket(
+	ctx sdk.Context,
+	markets []mmtypes.Market,
+) (bool, string) {
 	// Grab all the perpetuals markets
 	perps := d.perpKeeper.GetAllPerpetuals(ctx)
-	perpsMap := make(map[string]perpetualstypes.PerpetualMarketType)
+	restrictedMap := make(map[string]bool, len(perps))
```

Comment:
Why do you set the length here? In the unusual case where all perps are cross and are added to the restrictedMap and we add USDT/USD (which doesn't have a perp), this would cause issues

Reply:
Nvm i reread the documentation for make on maps and the length arg just sets the initial space